components/screens/ShoppingCart.js

```diff
@@ -1,13 +1,47 @@
-import React from 'react';
+import React, {useState} from 'react';
+import { Button } from 'react-native';
 import { View, ScrollView, Text, StyleSheet } from 'react-native';
 import NavBar from '../NavBar';
 
-function ShoppingCart({navigation}) {
+function ShoppingCart({ screenProps,navigation }) {
+
+    const [shoppingCart, setShoppingCart] = useState(screenProps.shoppingCart);
+    
+    const createList = () => {
+        return shoppingCart.map((element) => {
+            return (
+                <View style={{margin:10}}>
+                    <Text>{element.name}</Text>
+                    <Text>{element.price}</Text>
+                    <Button title='remove' onPress={()=>removeShoppingCartItem(element.key)}></Button>
+                </View>
+            );
+        });
+    };
+
+    const [list, setList] = useState(createList());
+
+    const removeShoppingCartItem = (itemKey) =>{
+        alert('Item Removed');
+
+        let updatedList = screenProps.shoppingCart
+        for(let index = 0; index < updatedList.length; index++){
+            if(updatedList[index].key == itemKey){
+                updatedList.splice(index,1);
```

Comment:
why `splice` ?

`splice` is mutating method and you are directly performing the operation on the direct array.
tried `slice` ?